core/src/display_object/movie_clip.rs

```diff
@@ -4132,6 +4132,18 @@ impl<'gc, 'a> MovieClip<'gc> {
                                 if self.avm2_class().is_none() {
                                     self.set_avm2_class(activation.gc(), Some(class_object));
                                 }
+
+                                // We also need to register this MovieClip as a character now
+
+                                let instantiated =
```

Comment:
what would have happened if we used the same `self` directly, instead of making a copy?